repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -383,6 +384,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             mutableHttpResponse.sendError(HttpServletResponse.SC_BAD_GATEWAY, "Error processing request");
             mutableHttpResponse.setLastException(ex);
         } finally {
+            try {
+                boolean isValidMethod = false;
+                for (HttpComponentFactory hcf : HttpComponentFactory.values()) {
+                    if (hcf.name().equalsIgnoreCase(mutableHttpRequest.getMethod())) {
+                        isValidMethod = true;
+                    }
+                }
+                if (!isValidMethod) {
+                    throw new Exception();
```

Comment:
You probably don't need to use a try catch with an exception here. You already know the case, so just do the work instead of handling it as an exception.